server/controllers/profile.js

```diff
@@ -1,47 +1,50 @@
 const mongoose = require('mongoose');
 const asyncHandler = require('express-async-handler');
 
+const User = require('../models/User');
 const Profile = require('../models/Profile');
 
-// @route GET /profile
+// @route GET /profiles
 //Search for all profiles
 exports.searchProfiles = asyncHandler(async (req,res,next) => {
-    try {
-        const profiles = await Profile.find();
-        res.status(200).json({
-          success: {
-            profiles
-          }
-        });
-      } catch (error) {
-        res.status(500);
-        throw new Error(error.message);
+  try {
+    const users = await User
+      .find({})
+      .populate({ path: "profile", match: { isDogSitter: { $eq: true }, price: {$exists: true}, city: {$exists: true}}})
+      .select("-password");
+    const userProfiles = users.filter(user => user.profile != null);
+    res.status(200).json({
+      success: {
+        users: userProfiles,
       }
-})
-
-// @route POST /profile
-//Create a new profile
-exports.createProfile = asyncHandler(async (req,res,next) => {
-    const profile = req.body;
-    const newProfile = new Profile({
-        ...profile
     });
+  } catch (error) {
+    res.status(500);
+    throw new Error(error.message);
+  }
+})
 
-    try {
-        await newProfile.save();
-        res.status(201).json({
-          success: {
-            profile: newProfile
-          }
-        });
-        
-    } catch (error) {
-        res.status(500);
-        throw new Error(error.message);
+// @route GET /profiles
+//Search for all profiles excluding current user
+exports.searchProtectedProfiles = asyncHandler(async (req,res,next) => {
+  try {
+    const users = await User
+      .find({_id : {$ne: req.user.id}})
+      .populate({ path: "profile", match: { isDogSitter: { $eq: true }, price: {$exists: true}, city: {$exists: true}}})
+      .select("-password");
+    const userProfiles = users.filter(user => user.profile != null);
+    res.status(200).json({
+      success: {
+        users: userProfiles,
       }
+    });
+  } catch (error) {
+    res.status(500);
+    throw new Error(error.message);
+  }
```

Comment:
this is overkill. We can just hit the regular route to fetch users, and check if `req.user.id` exists. If it does, exclude it

Reply:
Initially, I was trying to use one route but on excluding the `protect` middleware (to make the route accessible to all users), the JWT token from the FE is not decoded as a result `req.user = undefined` on all requests. Hence, I created two routes.

Currently, I have conditionally applied the `protect` middleware, see below:

![Capture](https://user-images.githubusercontent.com/75184965/119761552-d84d1400-be60-11eb-8859-09e0d1d820a0.PNG)

Thanks for pointing this out! :)